pkg/controller/jobframework/validation.go

```diff
@@ -144,3 +145,19 @@ func validateUpdateForMaxExecTime(oldJob, newJob GenericJob) field.ErrorList {
 	}
 	return nil
 }
+
+// ValidateImmutablePodSpec function is used for serving workloads to ensure no changes are allowed
+// to the PodSpec except for the image field in containers.
+func ValidateImmutablePodSpec(newPodSpec *corev1.PodSpec, oldPodSpec *corev1.PodSpec, fieldPath *field.Path) field.ErrorList {
+	newPodSpec = newPodSpec.DeepCopy()
+	oldPodSpec = oldPodSpec.DeepCopy()
+	resetPodSpecMutableFields(newPodSpec)
+	resetPodSpecMutableFields(oldPodSpec)
+	return apivalidation.ValidateImmutableField(newPodSpec, oldPodSpec, fieldPath)
+}
+
+func resetPodSpecMutableFields(podSpec *corev1.PodSpec) {
+	for i := range podSpec.Containers {
+		podSpec.Containers[i].Image = ""
+	}
+}
```

Comment:
Let's move the code to LWS for now as it is only used there. Also I don't like the function has a side effect. Please follow the pattern from core k8s https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/core/validation/validation.go#L5275.
It is ok to deep copy one of them. We will improve performance if needed.

Reply:
I synced with @mbobrovskyi that the function can be reused for STS. In that case I'm ok to keep it here. 
I still would like to avoid having the side effect of updating the specs.

Reply:
Done